prbench/src/prbench/envs/geom3d/packing3d.py

```diff
@@ -0,0 +1,765 @@
+"""Environment where multiple objects must be packed into a rack without collisions."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Type as TypingType
+
+import numpy as np
+import pybullet as p
+from pybullet_helpers.geometry import Pose, set_pose
+from pybullet_helpers.inverse_kinematics import check_body_collisions
+from pybullet_helpers.utils import (
+    create_pybullet_block,
+    create_pybullet_triangle,
+    get_triangle_vertices,
+)
+from relational_structs import Object, ObjectCentricState, Type
+from relational_structs.utils import create_state_from_dict
+
+from prbench.core import ConstantObjectPRBenchEnv, FinalConfigMeta
+from prbench.envs.geom3d.base_env import (
+    Geom3DEnvConfig,
+    ObjectCentricGeom3DRobotEnv,
+)
+from prbench.envs.geom3d.object_types import (
+    Geom3DCuboidType,
+    Geom3DEnvTypeFeatures,
+    Geom3DRobotType,
+    Geom3DTriangleType,
+)
+from prbench.envs.geom3d.utils import Geom3DObjectCentricState, is_on_top
+from prbench.envs.utils import PURPLE
+
+
+@dataclass(frozen=True)
+class Packing3DEnvConfig(Geom3DEnvConfig, metaclass=FinalConfigMeta):
+    """Config for Packing3DEnv()."""
+
+    # Table.
+    table_pose: Pose = Pose((0.3, 0.0, -0.175))
+    table_rgba: tuple[float, float, float, float] = (0.5, 0.5, 0.5, 1.0)
+    table_half_extents: tuple[float, float, float] = (0.2, 0.4, 0.25)
+
+    # rack (target) region.
```

Comment:
nit: captalize

Reply:
not resolved! (this one's not important, but I want to stay in the habit of resolving or responding to all comments)